src/EditorFeatures/Core/SymbolSearch/SymbolSearchUpdateEngineFactory.cs

```diff
@@ -36,144 +39,63 @@ private class RemoteUpdateEngine : ISymbolSearchUpdateEngine
-                        // client can be null if host is shutting down
-                        _client.ConnectionChanged += OnConnectionChanged;
-                    }
-                }
-            }
-
-            private async Task<RemoteHostClient.Session> TryGetSessionAsync()
-            {
-                using (await _gate.DisposableWaitAsync(_cancellationToken).ConfigureAwait(false))
-                {
-                    if (_sessionDoNotAccessDirectly != null)
-                    {
-                        return _sessionDoNotAccessDirectly;
-                    }
-
-                    if (_client == null)
-                    {
-                        // client can be null if host is shutting down
-                        return null;
-                    }
-
-                    // We create a single session and use it for the entire lifetime of this process.
-                    // That single session will be used to do all communication with the remote process.
-                    // This is because each session will cause a new instance of the RemoteSymbolSearchUpdateEngine
-                    // to be created on the remote side.  We only want one instance of that type.  The
-                    // alternative is to make that type static variable on the remote side.  But that's
-                    // much less clean and would make some of the state management much more complex.
-                    _sessionDoNotAccessDirectly = await _client.TryCreateServiceSessionAsync(
-                        WellKnownServiceHubServices.RemoteSymbolSearchUpdateEngine,
-                        _logService,
-                        _cancellationToken).ConfigureAwait(false);
-
-                    return _sessionDoNotAccessDirectly;
-                }
+                _session = session;
             }
 
             public async Task<ImmutableArray<PackageWithTypeResult>> FindPackagesWithTypeAsync(
                 string source, string name, int arity)
             {
-                var session = await TryGetSessionAsync().ConfigureAwait(false);
-                if (session == null)
+                var (success, results) = await _session.TryInvokeAsync<ImmutableArray<PackageWithTypeResult>>(
+                    nameof(IRemoteSymbolSearchUpdateEngine.FindPackagesWithTypeAsync),
+                    source, name, arity).ConfigureAwait(false);
+                if (!success)
                 {
-                    // we couldn't get session. most likely remote host is gone
+                    // keep alive session couldn't run. most likely remote host is gone
                     return ImmutableArray<PackageWithTypeResult>.Empty;
                 }
 
-                var results = await session.InvokeAsync<ImmutableArray<PackageWithTypeResult>>(
-                    nameof(IRemoteSymbolSearchUpdateEngine.FindPackagesWithTypeAsync),
-                    source, name, arity).ConfigureAwait(false);
-
                 return results;
```

Comment:
this is a very strange pattern.  All the callers get a boolean and a result value.  but if the boolean is false, they just return basically the default value.  i think i'd prefer if we just had InvokeAsync just return the default value, and use ?? or .NullOrEmpty on hte result to convert it as necessary.

i.e. no one really does anything interesting when success=false so i don't see a lot of value in having this strange different API here.